Changing to use spring jdbc template

zhljen opened the pull request:
Changing hive connector to bootstrap use spring 
Changing the mysql based query runner to use use spring jdbc template

tgianos commented on metacat-common-server/src/main/java/com/netflix/metacat/common/server/package-info.java:
```diff
@@ -21,4 +21,8 @@
  * @author amajumdar
  */
+
+@ParametersAreNonnullByDefault
```
Thanks for doing a lot of these!

tgianos commented on metacat-common-server/src/main/java/com/netflix/metacat/common/server/usermetadata/LookupService.java:
```diff
@@ -33,31 +34,39 @@
      * @return lookup
      */
-    Lookup get(String name);
+    default Lookup get(String name) {
```
Make all these final

zhljen replied:
Agreed

tgianos commented on metacat-common-server/src/main/java/com/netflix/metacat/common/server/usermetadata/LookupService.java:
```diff
@@ -33,31 +34,39 @@
      * @return scalar lookup value
      */
-    String getValue(String name);
+    default String getValue(String name) {
```
You should probably flag the return of these as `@Nullable` so calling methods are prepared. That or we should throw "Unsupported operation"

tgianos commented on metacat-common-server/src/main/java/com/netflix/metacat/common/server/util/DataSourceManager.java:
```diff
@@ -1,40 +1,55 @@
     private static DataSourceManager instance = new DataSourceManager();
     private Map<String, DataSource> dataSources = Maps.newConcurrentMap();
 
+    @Autowired
```
I'd recommend against using setter injection if at all possible. It makes testing a little harder and makes it so taht after construction this class isn't really fully ready to be used.

zhljen replied:
ok.

tgianos commented on metacat-common-server/src/main/java/com/netflix/metacat/common/server/util/DataSourceManager.java:
```diff
@@ -108,6 +123,47 @@ private synchronized void createDataSource(final String catalogName, final Map p
+     * @throws Exception exception
+     */
+    @PostConstruct
+    public void init() throws Exception {
```
We can talk about this but I don't quite understand why you don't just use this class and Inject it into the Mysql configuration classes and then just modify this within them. This seems backwards.

zhljen replied:
This is being also used in hive, tag service, lookup service. Let's talk about this.

tgianos commented on metacat-common-server/src/main/java/com/netflix/metacat/common/server/util/JdbcUtil.java:
```diff
@@ -0,0 +1,54 @@
+ * @since 1.1.0
+ */
+public class JdbcUtil {
+    @Getter
```
FYI if you mark a field as `final` and then put `@Data` or these other annotations on the class Lombok will not generate setters. It's cleaner than this. You just have to make sure the final members are instantiated during construction.

zhljen replied:
Think we don't need the other stuff generated by @Data, such as equalTo, toString. We might need them later, I will add the @Data annotation.

tgianos commented on metacat-common-server/src/main/java/com/netflix/metacat/common/server/util/JdbcUtil.java:
```diff
@@ -0,0 +1,54 @@
+     *
+     * @param dataSource data source
+     */
+    public JdbcUtil(@Nullable final DataSource dataSource) {
```
Why would anyone instantiate this class with a null datasource?

tgianos commented on metacat-common-server/src/main/java/com/netflix/metacat/common/server/util/MetacatConnectorProperties.java:
```diff
@@ -0,0 +1,46 @@
+@Data
+@NoArgsConstructor
+@Component
+public class MetacatConnectorProperties {
```
1. Rename this class to something like `ConnectorConfig`. Metacat is redundant and properties is specific as we also pass in other objects and such
2. This is a DTO and as such should be completely immutable after construction. Please make all members `private final` and either provide an all args constructor or a builder to create the object. Anything required can't be nullable so if you use the builder pattern the builder constructor should take those fields as parameters.

zhljen replied:
Changed as suggested..

tgianos commented on metacat-common-server/src/main/java/com/netflix/metacat/common/server/util/ProfileConstants.java:
```diff
@@ -0,0 +1,31 @@
+ * @author zhenl
+ * @since 1.1.0
+ */
+public final class ProfileConstants {
```
I think we talked about this in person but I don't like the idea of baking "profiles" which is a configuration construct into the code like this. It says something is wrong in the architecture of the code. We should fix the problem causing the need for this class and get rid of it.

tgianos commented on metacat-common-server/src/main/java/com/netflix/metacat/common/server/util/ThreadServiceManager.java:
```diff
@@ -37,7 +36,7 @@
-// TODO: Move to Hive connector
+@Component
 public class ThreadServiceManager {
     private ListeningExecutorService executor;
```
This should be immutable so lets make it private final.

zhljen replied:
agreed.

tgianos commented on metacat-common-server/src/main/java/com/netflix/metacat/common/server/util/ThreadServiceManager.java:
```diff
@@ -47,7 +46,7 @@
      * @param config Program configuration
      */
     @Inject
```
Should be `@Autowired`

tgianos commented on metacat-common/src/main/java/com/netflix/metacat/common/json/MetacatJsonException.java:
```diff
@@ -17,10 +17,28 @@
  * Metacat JSON utility related exception.
  */
 public class MetacatJsonException extends RuntimeException {
+    @Getter
```
Annotations should be on the class that way new members get it also. Also this should be immutable so there's no reason to have a setter.

tgianos commented on metacat-connector-hive/src/main/java/com/netflix/metacat/connector/hive/HiveConnectorConfig.java:
```diff
@@ -0,0 +1,180 @@
+ * @author zhenl
+ * @since 1.1.0
+ */
+@Configuration
```
We should talk about this. I think it's going to be a problem due to classpath scanning. Boot by default launches with classpath scanning for everything under `com.netflix.metacat` so this class will be picked up by the main spring app context when what we really want is it to only be picked up when a connector is instantiated on demand. You shouldn't have to mark all the beans as `prototype` as you'd want them to be singletons within that connector instance spring context. In this current manner every time a bean is requested it will create a new one which is not desirable. I'm not currently sure the best way to do this but we can research it.

tgianos commented on metacat-connector-hive/src/main/java/com/netflix/metacat/connector/hive/HiveConnectorConfig.java:
```diff
@@ -0,0 +1,180 @@
+    @Bean(name = "hivePartitionService")
+    @Scope("prototype")
+    public HiveConnectorPartitionService partitionService(
+        @Value("testhive") final String catalogName,
```
why are you baking in the concept of `testhive` here?

tgianos commented on metacat-connector-hive/src/main/java/com/netflix/metacat/connector/hive/HiveConnectorConfig.java:
```diff
@@ -0,0 +1,180 @@
+     * @param hiveMetacatConverter metacat converter
+     * @return HiveConnectorPartitionService
+     */
+    @Bean(name = "hivePartitionService")
```
Is there a reason you're explicitly naming the beans as opposed to just relying on the method names? There should only be one partition service right?

tgianos commented on metacat-connector-hive/src/main/java/com/netflix/metacat/connector/hive/HiveConnectorConfig.java:
```diff
@@ -0,0 +1,180 @@
+    @Bean(name = "fastHivePartitionService")
+    @Scope("prototype")
+    public HiveConnectorFastPartitionService fastHivePartitionService(
+        @Value("testhive") final String catalogName,
```
I think what you mean here instead of `@Value` is something like `@Qualified("nameOfBean")`

tgianos commented on metacat-connector-hive/src/main/java/com/netflix/metacat/connector/hive/HiveConnectorFactory.java:
```diff
@@ -95,18 +93,37 @@ public HiveConnectorFactory(
-        final Module hiveModule = new HiveConnectorModule(config,
-            catalogName, configuration, infoConverter, client, registry);
-        this.injector = Guice.createInjector(hiveModule);
+        final ApplicationContext ctx = new AnnotationConfigApplicationContext(HiveConnectorConfig.class);
```
Here is where I think you should add the connector name and other things as beans and then call `refresh` on the context.

tgianos commented on metacat-connector-hive/src/main/java/com/netflix/metacat/connector/hive/HiveConnectorFastTableService.java:
```diff
@@ -51,16 +45,17 @@
  */
 @Slf4j
+@Transactional(readOnly = true)
 public class HiveConnectorFastTableService extends HiveConnectorTableService {
```
As I read through this PR an idea is forming in my head that what we probably need to do in the main application is flag the `@ComponentScan` on the `@SpringBootApplication` class to exclude packages. Then we make our connectors all under the packages to exclude. Something like `com.netflix.metacat.connector.impl.{hive,redshift,ect}`. That way these are isolated.

tgianos commented on metacat-main/src/main/java/com/netflix/metacat/main/configs/ServicesConfig.java:
```diff
@@ -79,13 +83,50 @@ public UserMetadataService userMetadataService(
+     * @return User metadata service based on MySql
+     */
+    @Bean
+    @ConditionalOnMissingBean
```
Probably need to say `type = TagService.class` in this annotation otherwise I think it just looks for a missing bean of name `tagService`

tgianos commented on metacat-main/src/main/java/com/netflix/metacat/main/configs/ServicesConfig.java:
```diff
@@ -79,13 +83,50 @@ public UserMetadataService userMetadataService(
      * @return Lookup service backed by MySQL
      */
     @Bean
+    @Profile({ProfileConstants.MYSQL_USERMETADATA})
```
These beans should not be in this file they should be defined in a mysql implementation module. Lets talk about this in team meeting.

zhljen commented on gradle/wrapper/gradle-wrapper.properties:
```diff
@@ -1,6 +1,6 @@
 zipStoreBase=GRADLE_USER_HOME
 zipStorePath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-3.5-bin.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-3.5-all.zip
```
agreed.

zhljen commented on metacat-common-server/build.gradle:
```diff
@@ -40,7 +40,8 @@ dependencies {
     compile("org.springframework.boot:spring-boot-starter-logging")
     compile("org.springframework.cloud:spring-cloud-starter-spectator")
-
+    compile("org.springframework:spring-jdbc")
```
you mean 'spring-boot-starter-jdbc' , which seems not including the jdbc-template classes, and I have to explicitly depend on spring-jdbc

tgianos commented on metacat-common-server/src/main/java/com/netflix/metacat/common/server/util/DataSourceManager.java:
```diff
@@ -30,15 +29,28 @@
+     * @param environment environment
+     */
+    @Autowired
+    public void setEnvironment(final Environment environment) {
```
I think there is a Spring interface `EnvironmentAware` or something like that which you can implement to have spring inject the Environment. Still think we need to sit down and fully redesign this class so it's not doing setter injection.

tgianos commented on metacat-common-server/src/main/java/com/netflix/metacat/common/server/util/MetacatConnectorConfig.java:
```diff
@@ -0,0 +1,45 @@
+     */
+    private final String catalogName;
+    /**
+     * metacata config.
```
Spelling.

tgianos commented on metacat-common-server/src/main/java/com/netflix/metacat/common/server/util/MetacatConnectorConfig.java:
```diff
@@ -0,0 +1,45 @@
+ */
+@AllArgsConstructor
+@Data
+public class MetacatConnectorConfig {
```
Naming this `MetacatConnectorConfig` is kind of redudant no? Obviously everything in our code base is Metacat haha.. I'd just go with `ConnectorConfig` but that's my opinion

tgianos commented on metacat-common-server/src/main/java/com/netflix/metacat/common/server/util/ThreadServiceManager.java:
```diff
@@ -37,17 +35,16 @@
 @Slf4j
-// TODO: Move to Hive connector
+@Component
 public class ThreadServiceManager {
```
Do we use this class in the main Application at all or can it be moved to Hive as the TODO said?

I know the spring context already has an `ExecutorService` so creating yet another one in here is redundanta nd not managed by spring (e.g. auto closed)

tgianos commented on metacat-common-server/src/main/java/com/netflix/metacat/common/server/util/ThreadServiceManager.java:
```diff
@@ -37,17 +35,16 @@
      */
-    @Inject
-    public ThreadServiceManager(@Nonnull @NonNull final Config config) {
+    public ThreadServiceManager(final Config config) {
```
If you mark the class as `@Component` why don't you have `@Autowired` here?

tgianos commented on metacat-common/src/main/java/com/netflix/metacat/common/json/MetacatJsonException.java:
```diff
@@ -17,10 +17,28 @@
  */
+@Getter
+@Setter
 public class MetacatJsonException extends RuntimeException {
```
There's really no reason for an exception class to have a setter. It's immutable after being thrown. Make the input JSON final.

tgianos commented on metacat-connector-hive/src/main/java/com/netflix/metacat/connector/hive/HiveConnectorConfig.java:
```diff
@@ -0,0 +1,258 @@
+ *          limitations under the License.
+ */
+
+package com.netflix.metacat.connector.hive;
```
I still think we need to move all our connector packages under something like `com.netflix.metacat.connector.impl.*` so that we can safely ignore that package. Either that or ensure that we don't have anything in the `com.netflix.metacat.connector` package we want scanned by default. Which we should be able to do cause it's in common-server and we probably shouldn't auto wire anything that package anyway

tgianos commented on metacat-connector-hive/src/main/java/com/netflix/metacat/connector/hive/HiveConnectorConfig.java:
```diff
@@ -0,0 +1,258 @@
+     * @return HiveConnectorTableService
+     */
+    @Bean
+    @ConditionalOnProperty(value = "useHiveFastService", havingValue = "false")
```
This seems like you want this to be the default and then you want to override it when the property is true. In this case it might be cleaner to say `@ConditionalOnMissingBean(HiveConnectorTableService.class)`. This way it's a catch all for any other case we might add later for implementation of these services. Example: Lets say one day we have a HiveConnectorTableService implementation that's not fast and not the default but then we start enabling properties for it. Well now this condition needs to become a compound conditional and each time we add a new one a new condition. It doesn't scale as cleanly as just being a catch all.

tgianos commented on metacat-connector-hive/src/main/java/com/netflix/metacat/connector/hive/HiveConnectorConfig.java:
```diff
@@ -0,0 +1,258 @@
+     * @return IMetacatHiveClient
+     * @throws Exception exception
+     */
+
```
Weird extra return here.

tgianos commented on metacat-connector-hive/src/main/java/com/netflix/metacat/connector/hive/HiveConnectorConfig.java:
```diff
@@ -0,0 +1,258 @@
+     * @return data source
+     */
+    @Bean
+    @ConditionalOnProperty(value = "useHiveFastService", havingValue = "true")
```
You have this conditional a lot. I think what you should do in this case is create a whole new `Config` class which the whole class is conditional on this property being true and wire all the beans you need in there. Leave this config class as the default. That will be much cleaner and easier to understand what's going on.

tgianos commented on metacat-connector-hive/src/main/java/com/netflix/metacat/connector/hive/HiveConnectorFastTableService.java:
```diff
@@ -51,16 +46,17 @@
  * @since 1.0.0
  */
 @Slf4j
+@Transactional(readOnly = true)
```
@ajoymajumdar please look at all the transactional stuff.

zhljen commented:
On Thu, Jun 15, 2017 at 9:55 AM, Tom Gianos <notifications@github.com>
wrote:

> *@tgianos* requested changes on this pull request.
> ------------------------------
>
> In metacat-common-server/src/main/java/com/netflix/metacat/
> common/server/util/DataSourceManager.java
> <https://github.com/Netflix/metacat/pull/144#discussion_r122251541>:
>
> >
>      private DataSourceManager() {
>      }
>
>      /**
> +     * Set the environment instance.
> +     *
> +     * @param environment environment
> +     */
> +    @Autowired
> +    public void setEnvironment(final Environment environment) {
>
> I think there is a Spring interface EnvironmentAware or something like
> that which you can implement to have spring inject the Environment. Still
> think we need to sit down and fully redesign this class so it's not doing
> setter injection.
>
Actually, this environment variable should be removed. I have removed it.


> ------------------------------
>
> In metacat-common-server/src/main/java/com/netflix/metacat/
> common/server/util/MetacatConnectorConfig.java
> <https://github.com/Netflix/metacat/pull/144#discussion_r122251742>:
>
> > +import lombok.Data;
> +
> +import java.util.Map;
> +
> +/**
> + * Connector Config.
> + */
> +@AllArgsConstructor
> +@Data
> +public class MetacatConnectorConfig {
> +    /**
> +     * catalog name.
> +     */
> +    private final String catalogName;
> +    /**
> +     * metacata config.
>
> Spelling.
>
Fixed.

> ------------------------------
>
> In metacat-common-server/src/main/java/com/netflix/metacat/
> common/server/util/MetacatConnectorConfig.java
> <https://github.com/Netflix/metacat/pull/144#discussion_r122251867>:
>
> > +
> +package com.netflix.metacat.common.server.util;
> +
> +import com.netflix.metacat.common.server.properties.Config;
> +import com.netflix.spectator.api.Registry;
> +import lombok.AllArgsConstructor;
> +import lombok.Data;
> +
> +import java.util.Map;
> +
> +/**
> + * Connector Config.
> + */
> +@AllArgsConstructor
> +@Data
> +public class MetacatConnectorConfig {
>
> Naming this MetacatConnectorConfig is kind of redudant no? Obviously
> everything in our code base is Metacat haha.. I'd just go with
> ConnectorConfig but that's my opinion
>
Renamed.

>
>
------------------------------
>
> In metacat-common-server/src/main/java/com/netflix/metacat/
> common/server/util/ThreadServiceManager.java
> <https://github.com/Netflix/metacat/pull/144#discussion_r122252234>:
>
> > @@ -37,17 +35,16 @@
>   */
>  @Getter
>  @Slf4j
> -// TODO: Move to Hive connector
> +@Component
>  public class ThreadServiceManager {
>
> Do we use this class in the main Application at all or can it be moved to
> Hive as the TODO said?
>
> I know the spring context already has an ExecutorService so creating yet
> another one in here is redundanta nd not managed by spring (e.g. auto
> closed)
>
This is also being used in the metadata service. We discussed this earlier
that the spring ExecutorService is not as stable and feature rich as the
guava one we currently use. There're certain functionalities such as
Future.asList ( heavily used in elasticSearch ) are not supported. We
decided to stick to current one.


> ------------------------------
>
> In metacat-common-server/src/main/java/com/netflix/metacat/
> common/server/util/ThreadServiceManager.java
> <https://github.com/Netflix/metacat/pull/144#discussion_r122252446>:
>
> >
>      /**
>       * Constructor.
>       *
>       * @param config Program configuration
>       */
> -    @Inject
> -    public ThreadServiceManager(@Nonnull @NonNull final Config config) {
> +    public ThreadServiceManager(final Config config) {
>
> If you mark the class as @Component why don't you have @Autowired here?
>
I added it.

> ------------------------------
>
> In metacat-common/src/main/java/com/netflix/metacat/common/
> json/MetacatJsonException.java
> <https://github.com/Netflix/metacat/pull/144#discussion_r122252591>:
>
> >  /**
>   * Metacat JSON utility related exception.
>   */
> +@Getter
> +@Setter
>  public class MetacatJsonException extends RuntimeException {
>
> There's really no reason for an exception class to have a setter. It's
> immutable after being thrown. Make the input JSON final.
>
The usage of this exception may not include the input json. I have created
a new class MetacatJsonParseException, which will dedicate to the json
string parsing exception and it includes the input string as a final
variable.


> ------------------------------
>
> In metacat-connector-hive/src/main/java/com/netflix/metacat/
> connector/hive/HiveConnectorConfig.java
> <https://github.com/Netflix/metacat/pull/144#discussion_r122253184>:
>
> > @@ -0,0 +1,258 @@
> +/*
> + *       Copyright 2017 Netflix, Inc.
> + *          Licensed under the Apache License, Version 2.0 (the "License");
> + *          you may not use this file except in compliance with the License.
> + *          You may obtain a copy of the License at
> + *              http://www.apache.org/licenses/LICENSE-2.0
> + *          Unless required by applicable law or agreed to in writing, software
> + *          distributed under the License is distributed on an "AS IS" BASIS,
> + *          WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
> + *          See the License for the specific language governing permissions and
> + *          limitations under the License.
> + */
> +
> +package com.netflix.metacat.connector.hive;
>
> I still think we need to move all our connector packages under something
> like com.netflix.metacat.connector.impl.* so that we can safely ignore
> that package. Either that or ensure that we don't have anything in the
> com.netflix.metacat.connector package we want scanned by default. Which
> we should be able to do cause it's in common-server and we probably
> shouldn't auto wire anything that package anyway
>
Yes, we will filter out the whole connector package
com.netflix.metacat.connector.*

> ------------------------------
>
> In metacat-connector-hive/src/main/java/com/netflix/metacat/
> connector/hive/HiveConnectorConfig.java
> <https://github.com/Netflix/metacat/pull/144#discussion_r122254020>:
>
> > +        final MetacatConnectorConfig connectorConfig) {
> +        return new HiveConnectorDatabaseService(connectorConfig.getCatalogName(),
> +            metacatHiveClient, hiveMetacatConverter);
> +    }
> +
> +    /**
> +     * create hive connector table service.
> +     *
> +     * @param metacatHiveClient            metacat hive client
> +     * @param hiveMetacatConverters        hive metacat converters
> +     * @param hiveConnectorDatabaseService hive database service
> +     * @param metacatConnectorConfig       connector config
> +     * @return HiveConnectorTableService
> +     */
> +    @Bean
> +    @ConditionalOnProperty(value = "useHiveFastService", havingValue = "false")
>
> This seems like you want this to be the default and then you want to
> override it when the property is true. In this case it might be cleaner to
> say @ConditionalOnMissingBean(HiveConnectorTableService.class). This way
> it's a catch all for any other case we might add later for implementation
> of these services. Example: Lets say one day we have a
> HiveConnectorTableService implementation that's not fast and not the
> default but then we start enabling properties for it. Well now this
> condition needs to become a compound conditional and each time we add a new
> one a new condition. It doesn't scale as cleanly as just being a catch all.
>
Agree.

> ------------------------------
>
> In metacat-connector-hive/src/main/java/com/netflix/metacat/
> connector/hive/HiveConnectorConfig.java
> <https://github.com/Netflix/metacat/pull/144#discussion_r122254274>:
>
> > +            hiveConnectorDatabaseService, hiveMetacatConverters,
> +            Boolean.parseBoolean(
> +                metacatConnectorConfig.getConfiguration()
> +                    .getOrDefault(HiveConfigConstants.ALLOW_RENAME_TABLE, "false")),
> +            metacatConnectorConfig,
> +            dataSource);
> +    }
> +
> +    /**
> +     * create local hive client.
> +     *
> +     * @param metacatConnectorConfig connector config context
> +     * @return IMetacatHiveClient
> +     * @throws Exception exception
> +     */
> +
>
> Weird extra return here.
> ------------------------------
>
> In metacat-connector-hive/src/main/java/com/netflix/metacat/
> connector/hive/HiveConnectorConfig.java
> <https://github.com/Netflix/metacat/pull/144#discussion_r122254925>:
>
> > +                HMSHandlerProxy.getProxy(conf), metacatConnectorConfig.getRegistry());
> +        } catch (Exception e) {
> +            throw new IllegalArgumentException(
> +                String.format("Failed creating the hive metastore client for catalog: %s",
> +                    metacatConnectorConfig.getCatalogName()), e);
> +        }
> +    }
> +
> +    /**
> +     * hive DataSource.
> +     *
> +     * @param metacatConnectorConfig connector config.
> +     * @return data source
> +     */
> +    @Bean
> +    @ConditionalOnProperty(value = "useHiveFastService", havingValue = "true")
>
> You have this conditional a lot. I think what you should do in this case
> is create a whole new Config class which the whole class is conditional
> on this property being true and wire all the beans you need in there. Leave
> this config class as the default. That will be much cleaner and easier to
> understand what's going on.
>
Agree.

> ------------------------------
>
> In metacat-connector-hive/src/main/java/com/netflix/metacat/
> connector/hive/HiveConnectorFastTableService.java
> <https://github.com/Netflix/metacat/pull/144#discussion_r122255125>:
>
> > @@ -51,16 +46,17 @@
>   * @since 1.0.0
>   */
>  @Slf4j
> +@Transactional(readOnly = true)
>
> @ajoymajumdar <https://github.com/ajoymajumdar> please look at all the
> transactional stuff.
>
> —
> You are receiving this because you were assigned.
> Reply to this email directly, view it on GitHub
> <https://github.com/Netflix/metacat/pull/144#pullrequestreview-44352543>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AG4X8xKQSNs1Acjb0pcvmSXUmzUVbGo8ks5sEWITgaJpZM4N3Vjy>
> .
>

ajoymajumdar commented on metacat-common-server/src/main/java/com/netflix/metacat/common/server/connectors/ConnectorPlugin.java:
```diff
@@ -43,17 +38,13 @@
-        @Nonnull Map<String, String> configuration,
-        @Nonnull @NonNull Registry registry
+        String connectorName,
+        ConnectorConfig connectorConfig
```
Do we want to rename the class to ConnectorContext?

ajoymajumdar commented on metacat-common-server/src/main/java/com/netflix/metacat/common/server/usermetadata/UserMetadataService.java:
```diff
@@ -230,14 +235,6 @@ default int renameDefinitionMetadataKey(QualifiedName oldName, QualifiedName new
     }
 
     /**
-     * Start the user metadata service.
```
Why did we remove this?

ajoymajumdar commented on metacat-common-server/src/main/java/com/netflix/metacat/common/server/util/ConnectorConfig.java:
```diff
@@ -0,0 +1,45 @@
+ */
+@AllArgsConstructor
+@Data
+public class ConnectorConfig {
```
Can we rename this to ConnectorContext?

ajoymajumdar commented on metacat-common/src/main/java/com/netflix/metacat/common/json/MetacatJsonLocator.java:
```diff
@@ -113,7 +113,7 @@ public ObjectNode parseJsonObject(final String s) {
             return (ObjectNode) node;
         } else {
-            throw new MetacatJsonException("Cannot convert '" + s + "' to a json object");
+            throw new MetacatJsonParseException(s, "Cannot convert '" + s + "' to a json object");
```
What is the reason to add a class, instead of creating another constructor in MetacatJsonException
